### PR TITLE
Fix bc break of fp16

### PR DESCRIPTION
**Pull request by RyanXLi**
This PR:
- Fix bc break of fp16
- Added warning for importing deprecated mmdet version of fp16 utils
- refactored some import of fp16 utils now that mmcv 1.1.3 is up(and required by mmdet)

### CODE REVIEW

File: mmdet/core/fp16/deprecated_fp16_utils.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import warnings
+
+from mmcv.runner import (Fp16OptimizerHook, auto_fp16, force_fp32,
+                         wrap_fp16_model)
+
+
+class Depr_Fp16OptimizerHook(Fp16OptimizerHook):
+
```
Comment (Johnson-Wang) [inline mmdet/core/fp16/deprecated_fp16_utils.py:8]: Docstring.

**Review by Johnson-Wang — COMMENTED** [review]
It should be ready to go after adding some docstrings.

File: mmdet/core/fp16/deprecated_fp16_utils.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import warnings
+
+from mmcv.runner import (Fp16OptimizerHook, auto_fp16, force_fp32,
+                         wrap_fp16_model)
+
+
+class Depr_Fp16OptimizerHook(Fp16OptimizerHook):
+    """A wrapper class for the FP16 optimizer hook. This class wraps
+    Fp16OptimizerHook in "mmcv.runner" and shows a warning that the
```
Comment (ZwwWayne) [inline mmdet/core/fp16/deprecated_fp16_utils.py:9]: :class:`Fp16OptimizerHook`,  ``mmcv.runner``, ``mmdet.core`` might be better for docstring syntax.

File: mmdet/core/fp16/deprecated_fp16_utils.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import warnings
+
+from mmcv.runner import (Fp16OptimizerHook, auto_fp16, force_fp32,
+                         wrap_fp16_model)
+
+
+class Depr_Fp16OptimizerHook(Fp16OptimizerHook):
```
Comment (hellock) [inline mmdet/core/fp16/deprecated_fp16_utils.py:7]: Just use the complete word DeprecatedFp16OptimizerHook

**Review by Johnson-Wang — APPROVED** [review]

**Pull request merged**